src/commands/cmd_mount.rs

```diff
@@ -107,36 +109,105 @@ fn read_super_silent(path: &std::path::PathBuf) -> anyhow::Result<bch_sb_handle>
     bch_bindgen::sb_io::read_super_silent(&path, opts)
 }
 
-fn get_devices_by_uuid(uuid: Uuid) -> anyhow::Result<Vec<(PathBuf, bch_sb_handle)>> {
-    debug!("enumerating udev devices");
+fn device_property_map(dev: &udev::Device) -> HashMap<String, String> {
+    let rc: HashMap<_, _> = dev
+        .properties()
+        .map(|i| {
+            (
+                String::from(i.name().to_string_lossy()),
+                String::from(i.value().to_string_lossy()),
+            )
+        })
+        .collect();
+    rc
+}
+
+fn udev_bcachefs_info() -> anyhow::Result<HashMap<String, Vec<String>>> {
+    let mut info = HashMap::new();
     let mut udev = udev::Enumerator::new()?;
 
+    debug!("Walking udev db!");
+
     udev.match_subsystem("block")?;
+    udev.match_property("ID_FS_TYPE", "bcachefs")?;
 
-    let devs = udev
-        .scan_devices()?
-        .into_iter()
-        .filter_map(|dev| dev.devnode().map(ToOwned::to_owned))
-        .map(|dev| (dev.clone(), read_super_silent(&dev)))
-        .filter_map(|(dev, sb)| sb.ok().map(|sb| (dev, sb)))
-        .filter(|(_, sb)| sb.sb().uuid() == uuid)
-        .collect();
-    Ok(devs)
+    for dev in udev.scan_devices()? {
+        if !dev.is_initialized() {
+            continue;
+        }
+
+        let m = device_property_map(&dev);
+        if m.contains_key("ID_FS_UUID") && m.contains_key("DEVNAME") {
+            let fs_uuid = m["ID_FS_UUID"].clone();
+            let dev_node = m["DEVNAME"].clone();
+            info.insert(dev_node.clone(), vec![fs_uuid.clone()]);
+            info.entry(fs_uuid).or_insert(vec![]).push(dev_node.clone());
+        }
+    }
+
+    Ok(info)
 }
 
-fn get_uuid_for_dev_node(device: &std::path::PathBuf) ->  anyhow::Result<Option<Uuid>> {
+fn get_devices_by_uuid(
+    udev_bcachefs: &HashMap<String, Vec<String>>,
+    uuid: Uuid,
+) -> anyhow::Result<Vec<(PathBuf, bch_sb_handle)>> {
+    if !udev_bcachefs.is_empty() {
+        let uuid_string = uuid.hyphenated().to_string();
+
+        if let Some(devices) = udev_bcachefs.get(&uuid_string) {
+            let devs = devices
+                .iter()
+                .filter_map(|dev| {
+                    read_super_silent(&PathBuf::from(dev))
+                        .ok()
+                        .map(|sb| (PathBuf::from(dev), sb))
+                })
+                .collect();
+            Ok(devs)
+        } else {
+            Ok(Vec::new())
+        }
+    } else {
+        let mut udev = udev::Enumerator::new()?;
+        udev.match_subsystem("block")?;
+
+        let devs = udev
+            .scan_devices()?
+            .filter_map(|dev| dev.devnode().map(ToOwned::to_owned))
+            .map(|dev| (dev.clone(), read_super_silent(&dev)))
+            .filter_map(|(dev, sb)| sb.ok().map(|sb| (dev, sb)))
```

Comment:
We're doing a lot of similar-ish read_super_silent() loops. Maybe we could add a few one-two line helpers with more descriptive names to clean things up?

has_bcachefs_sb()
bcachefs_sb_uuid()

Reply:
I'll look into this, thanks